### PR TITLE
Fix `funk` and `funk name` scope for Ruby Class methods

### LINKED ISSUE #760: Scope `funk` doesn't match on Class methods - ruby
I was hoping to try and fix this myself as a way to get familiar with how cursorless + tree sitter works but it might take me a bit to get up to speed. Here is the issue:

Given this code:

```ruby
class Test
  class << self
    # `take funk` doesn't work
    def baz
    end
  end

  # `take funk` doesn't work
  def self.foo
    new.process
  end

  # `take funk` doesn't work
  def Test.bar
    new.process
  end

  # `take funk` works :D
  def process
    puts "process"
  end
end
```

Current use cases/scopes not covered:
- [x] `"change funk"` on a function defined `self.`
- [x] `"change funk"` on a function defined `ClassName.`
- [x] `"change funk"` on a function defined within `class << self .... end` block

### PR DESCRIPTION
**Pull request by richardmcmillen**
Tree sitter sees these as `singleton_methods` and this was missing from
the `namedFunction` and `functionName` definitions.

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)

---

Fixes #760

### CODE REVIEW

File: src/languages/ruby.ts [hunk 1]
```diff
@@ -167,8 +167,8 @@ const nodeMatchers: Partial<
   ifStatement: "if",
   functionCall: "call",
   comment: "comment",
-  namedFunction: "method",
-  functionName: "method[name]",
+  namedFunction: ["method", "singleton_method"],
+  functionName: ["method[name]", "singleton_method[name]"],
```
Comment (pokey) [inline src/languages/ruby.ts:171]: Please also add to `name` (line 187 below), and add a test for that. Just one test for that one is fine

**Review by pokey — COMMENTED** [review]
Looks good! See inline comment, and also

- [ ] Worth adding a test for "arg" within an arg list for one of these functions

I'll also give @Will-Sommers a chance to have a look

**Comment by richardmcmillen** [comment]
>Worth adding a test for "arg" within an arg list for one of these functions

I added a new test case which I think covers this if I understand correctly, and make the other change as well.

File: src/languages/ruby.ts (hunk 1 above)
Comment (Will-Sommers) [reply src/languages/ruby.ts:171]: Sorry I missed this suggestion in the issue @richardmcmillen!

**Review by Will-Sommers — APPROVED** [review]
lgtm, thank you!

**Review by pokey — APPROVED** [review]
Yep looks good to me too thanks!

**Pull request merged**